donazioni/viste.py

```diff
@@ -0,0 +1,259 @@
+from datetime import datetime
+from itertools import chain
+
+from django.core.urlresolvers import reverse
+from django.db.models import Q, Sum
+from django.shortcuts import redirect, get_object_or_404
+
+from anagrafica.permessi.applicazioni import RESPONSABILE_CAMPAGNA
+from anagrafica.permessi.costanti import GESTIONE_CAMPAGNE, GESTIONE_CAMPAGNA, COMPLETO, ERRORE_PERMESSI, MODIFICA
+from autenticazione.funzioni import pagina_privata
+from donazioni.forms import ModuloCampagna, ModuloEtichetta, ModuloFiltraCampagnePerEtichetta, ModuloDonazione, ModuloDonatore
+from donazioni.models import Campagna, Etichetta, Donatore, Donazione
+
+
+@pagina_privata
+def donazioni_home(request, me):
+    campagne = me.oggetti_permesso(GESTIONE_CAMPAGNA)
+    donatori = Donatore.objects.filter(donazioni__campagna__in=campagne)
+    fondi_raccolti = Donazione.objects.filter(campagna__in=campagne).aggregate(totale=Sum('importo'))
+    contesto = {
+        'sedi': me.oggetti_permesso(GESTIONE_CAMPAGNE),
+        'campagne': campagne,
+        'donatori': donatori,
+        'fondi_raccolti': fondi_raccolti['totale']
+    }
+    return 'donazioni.html', contesto
+
+
+@pagina_privata
+def campagne_elenco(request, me):
+    campagne = me.oggetti_permesso(GESTIONE_CAMPAGNA).annotate(totale_donazioni=Sum('donazioni__importo'))
+    modulo = ModuloFiltraCampagnePerEtichetta(request.POST or None, empty_permitted=True)
+    if modulo.is_valid() and modulo.cleaned_data['etichette']:
+        campagne = campagne.filter(etichette__in=modulo.cleaned_data['etichette'])
+    contesto = {
+        "campagne": campagne,
+        "puo_creare": me.ha_permesso(GESTIONE_CAMPAGNE),
+        "modulo_filtro": modulo,
+    }
+    return 'donazioni_campagne_elenco.html', contesto
+
+
+@pagina_privata(permessi=(GESTIONE_CAMPAGNE,))
+def campagna_nuova(request, me):
+
+    modulo = ModuloCampagna(request.POST or None, initial={"inizio": datetime.now()})
+
+    if modulo.is_valid():
+        campagna = modulo.save()
+        request.session['campagna_creata'] = campagna.pk
+        return redirect(campagna.url_responsabili)
+
+    contesto = {
+        "modulo": modulo
+    }
+    return 'donazioni_campagna_nuova.html', contesto
+
+
+@pagina_privata(permessi=(GESTIONE_CAMPAGNE,))
+def campagna_modifica(request, me, pk):
```

Comment:
Non vi e' alcun controllo sui permessi della campagna -- In questo modo qualunque responsabile per una qualunque campagna in un qualunque posto d'Italia, puo' modificare tutte le campagne in Gaia.